Documentation/network/lb-ipam.rst

```diff
@@ -0,0 +1,382 @@
+
+    $ kubectl get ippools
+    NAME        DISABLED   CONFLICTING   IPS AVAILABLE   AGE
+    blue-pool   false      false         254             25m
+    red-pool    false      true          254             11s
+
+The reason for the conflict is stated in the status and can be accessed like so
+
+.. code-block:: shell-session
+
+    $ kubectl get ippools/red-pool -o jsonpath='{.status.conflictReason}'
+    Pool conflicts since CIDR '20.0.10.0/24' overlaps CIDR '20.0.10.0/24' from IP Pool 'blue-pool'
+
+or
+
+.. code-block:: shell-session
+
+    $ kubectl describe ippools/red-pool
+    Name:         red-pool
+    #[...]
+    Status:
+      #[...]
+      Conflict Reason:  Pool conflicts since CIDR '20.0.10.0/24' overlaps CIDR '20.0.10.0/24' from IP Pool 'blue-pool'
+      Conflicting:      true
+      #[...]
+
+Disabling a Pool
+-----------------
+
+IP Pools can be disabled. Disabling a pool will stop LB IPAM from allocating
+new IPs from the pool, but doesn't remove existing allocations. This allows
+an administrator to slowly drain pool or reserve a pool for future use.
+
+.. code-block:: yaml
+
+    apiVersion: "cilium.io/v2alpha1"
+    kind: CiliumLoadBalancerIPPool
+    metadata:
+      name: "blue-pool"
+    spec:
+      cidrs:
+      - cidr: "20.0.10.0/24"
+      disabled: true
+
+.. code-block:: shell-session
+
+    $ kubectl get ippools          
+    NAME        DISABLED   CONFLICTING   IPS AVAILABLE   AGE
+    blue-pool   true       false         254             41m
+
+Status
+------
+
+The IP Pool's status contains additional counts which can be used to monitor
+the amount of used and available IPs. A machine parsable output can be gotten like so.
+
+.. code-block:: shell-session
+
+    $ kubectl get ippools -o jsonpath='{.items[*].status}' | jq
+    {
+      "cidrCounts": {
+        "20.0.10.0/24": {
+          "available": 254,
+          "total": 254,
+          "used": 0
+        },
+        "2004::/32": {
+          "available": 65534,
+          "total": 65534,
+          "used": 0
+        }
+      },
+      "conflictReason": "",
+      "conflicting": false,
+      "totalCounts": {
+        "available": 65788,
+        "total": 65788,
+        "used": 0
+      }
+    }
+
+Or human readable output like so
+
+.. code-block:: shell-session
+
+    $ kubectl describe ippools/blue-pool
+    Name:         blue-pool
+    Namespace:    
+    Labels:       <none>
+    Annotations:  <none>
+    API Version:  cilium.io/v2alpha1
+    Kind:         CiliumLoadBalancerIPPool
+    #[...]
+    Status:
+      Cidr Counts:
+        20.0.10.0/24:
+          Available:    254
+          Total:        254
+          Used:         0
+      Conflict Reason:  
+      Conflicting:      false
+      Total Counts:
+        Available:  254
+        Total:      254
+        Used:       0
+
+Services
+########
+
+Any service with ``.spec.type=LoadBalancer`` can get IPs from any pool as long
+as the IP Pool's service selector matches the service.
```

Comment:
What happens if multiple pools match a service? I assume one is selected at random.

Reply:
Yes. We do not explicitly have logic do pick a random Pool/cidr or to balance allocation, we simply pick the first we can match without additional guarantees.